### PR TITLE
Adding default script for bwctest.sh

### LINKED ISSUE #296: Add bwctest.sh in OpenSearch and one plugin
- [ ] OpenSearch Core - https://github.com/opensearch-project/OpenSearch/pull/1150
- [x] Anomaly Detection Plugin - https://github.com/opensearch-project/anomaly-detection/pull/189

### PR DESCRIPTION
**Pull request by saratvemulapalli**
### Description
Adding default script support for bwctests.
All repositories unless they need a custom file they would default to use `bwctest.sh`

Addressed follow up comments from https://github.com/opensearch-project/opensearch-build/pull/503
 
### Issues Resolved
Closes #296 
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: bundle-workflow/src/run_bwc_test.py [hunk 1]
```diff
@@ -10,19 +10,23 @@
...
     args = TestArgs()
+    console.configure(level=args.logging_level)
     with TemporaryDirectory(keep=args.keep) as work_dir:
         logging.info("Switching to temporary work_dir: " + work_dir)
-        os.chdir(work_dir)
-        bundle_manifest = BundleManifest.from_s3(
-            args.s3_bucket, args.build_id, args.opensearch_version, args.architecture, work_dir)
-        BwcTestSuite(bundle_manifest, work_dir, args.component, args.keep).execute()
+        with WorkingDirectory(work_dir) as cur_dir:
+            os.chdir(work_dir)
```
Comment (owaiskazi19) [inline bundle-workflow/src/run_bwc_test.py:26]: This won't be needed if you are using `WorkingDirectory `

Comment (saratvemulapalli) [reply bundle-workflow/src/run_bwc_test.py:26]: haha thanks for that. Bunch of conflicts and I missed this.

File: bundle-workflow/src/test_workflow/bwc_test/bwc_test_suite.py [hunk 2]
```diff
@@ -9,33 +9,43 @@
...
-        return output
+    def run_tests(self, work_dir, component_name):
+        script = self.script_finder.find_bwc_test_script(
+            component_name, work_dir
+        )
+        if os.path.exists(script):
+            cmd = f"{script}"
+            (status, stdout, stderr) = execute(cmd, work_dir, True, False)
+            return (status, stdout, stderr)
+        else:
```
Comment (dblock) [inline bundle-workflow/src/test_workflow/bwc_test/bwc_test_suite.py:39]: Is this possible? There's a default one, no?

**Review by dblock — APPROVED** [review]

File: bundle-workflow/src/test_workflow/bwc_test/bwc_test_suite.py (hunk 2 above)
Comment (saratvemulapalli) [reply bundle-workflow/src/test_workflow/bwc_test/bwc_test_suite.py:39]: Lol thats true. That was just incase to understand where things are failing. 
My likely a no-op.

Comment (dblock) [reply bundle-workflow/src/test_workflow/bwc_test/bwc_test_suite.py:39]: You should remove the `if`. Execute will fail if you don't have the script anyway.

**Review by dblock — DISMISSED** [review]

**Review by owaiskazi19 — DISMISSED** [review]

File: bundle-workflow/src/test_workflow/bwc_test/bwc_test_suite.py (hunk 2 above)
Comment (saratvemulapalli) [reply bundle-workflow/src/test_workflow/bwc_test/bwc_test_suite.py:39]: Sure, that makes sense. Let me quickly fix that.

**Review by owaiskazi19 — APPROVED** [review]

File: bundle-workflow/src/test_workflow/bwc_test/bwc_test_suite.py [hunk 3]
```diff
@@ -9,33 +9,38 @@
...
-    def run_tests(self, work_dir):
-        bwc_script = "bwctest.sh"
-        run_bwctests = f"./{bwc_script}"
-        output = subprocess.check_output(run_bwctests, cwd=work_dir, shell=True)
-        return output
+    def run_tests(self, work_dir, component_name):
+        script = self.script_finder.find_bwc_test_script(
+            component_name, work_dir
+        )
+        cmd = f"{script}"
```
Comment (dblock) [inline bundle-workflow/src/test_workflow/bwc_test/bwc_test_suite.py:35]: You don't need `cmd`, just pass `script` into `execute`.

File: bundle-workflow/src/test_workflow/bwc_test/bwc_test_suite.py [hunk 4]
```diff
@@ -9,33 +9,38 @@
...
+        (status, stdout, stderr) = execute(cmd, work_dir, True, False)
```
Comment (dblock) [inline bundle-workflow/src/test_workflow/bwc_test/bwc_test_suite.py:36]: `return execute(...)`

**Review by dblock — APPROVED** [review]

**Pull request merged**